apps/dashboard/src/components/contract-components/contract-publish-form/index.tsx

```diff
@@ -259,8 +273,16 @@ export const ContractPublishForm: React.FC<ContractPublishFormProps> = ({
   );
 
   const shouldShowDynamicFactoryInput = useMemo(
-    () => isPluginRouter || (isDynamicContract && hasExtensionsParam),
-    [isPluginRouter, isDynamicContract, hasExtensionsParam],
+    () =>
+      isPluginRouter ||
+      (isDynamicContract && hasExtensionsParam) ||
+      isModularContract,
+    [isPluginRouter, isDynamicContract, isModularContract, hasExtensionsParam],
```

Comment:
I wonder if we should have a function that does this logic in the SDK, seems like it could be generally useful

Reply:
++ @kumaryash90